addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -196,14 +230,24 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                this._stopLoading();
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                this._stopLoading();
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if (this.thread && this.threadCache.isLoading) {
+                if (!this._loaderTimeout) {
+                    this._prepareLoading();
+                }
+            } else {
+                this._stopLoading();
+            }
```

Comment:
looks like this part can be dropped

Reply:
Not convinced but I'll change, if we spot a glitch with that, we can bring it back easily :)